File: topgames/src/main/java/de/steamcharts/topgames/model/RankingData.java
package de.steamcharts.topgames.model;

import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(name="rankingdata")
public class RankingData {

    @EmbeddedId
    private RankingDataPK rankingDataPK;

    @Column(name="num_players")
    private int numPlayers;

    @Column(name="day_peak")
    private int dayPeak;

    @Column(name="ranking")
    private int ranking;

    public RankingDataPK getRankingDataPK() {
        return rankingDataPK;
    }

    public void setRankingDataPK(RankingDataPK rankingDataPK) {
        this.rankingDataPK = rankingDataPK;
    }

    public int getNumPlayers() {
        return numPlayers;
    }

    public void setNumPlayers(int numPlayers) {
        this.numPlayers = numPlayers;
    }

    public int getDayPeak() {
        return dayPeak;
    }

    public void setDayPeak(int dayPeak) {
        this.dayPeak = dayPeak;
    }

    public int getRanking() {
        return ranking;
    }

    public void setRanking(int ranking) {
        this.ranking = ranking;
    }
}

File: webscraper/main/ranking_data.py
from game import Game
import datetime

class RankingData:

    def __init__(self, date, game, num_players, day_peak, ranking):
        self.date: datetime = date
        self.game: Game = game
        self.num_players: int = num_players
        self.day_peak: int = day_peak
        self.ranking: int = ranking

    def __str__(self):
        return (f"""
            {self.game.name} : {type(self.game.name)} \n 
            {self.date} : {type(self.date)} \n
            {self.num_players} : {type(self.num_players)} \n
            {self.day_peak} : {type(self.day_peak)} \n
            {self.ranking} : {type(self.ranking)}
        """)